actioncable/lib/action_cable/server/configuration.rb

```diff
@@ -35,15 +35,16 @@ def pubsub_adapter
         rescue LoadError => e
           # We couldn't require the adapter itself. Raise an exception that
           # points out config typos and missing gems.
-          if e.path == path_to_adapter
+          # When `Gem::LoadError` occurs, possible that `path` is not set correctly. Instead can get the gem name from `name` attribute.
+          if (e.path == path_to_adapter) || (e.respond_to?(:name) && e.name == adapter)
```

Comment:
This condition intends to only pick up the "main" require from line 34; the new part sounds like it might match some problems occurring *inside* the gem?